app/Services/MentorService.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Services;
+
+use App\Repositories\MentorRepositoryInterface;
+
+class MentorService
+{
+    private MentorRepositoryInterface $mentorRepositoryInterface;
+
+    public function __construct(MentorRepositoryInterface $mentorRepositoryInterface)
+    {
+        $this->mentorRepositoryInterface = $mentorRepositoryInterface;
+    }
+
+    public function getAllMentor()
+    {
+        $searchAuthor = \request()->search;
```

Comment:
You must capture the request at controller then inject to service as an array parameter. Please don't use request facade

Reply:
fix